data-prepper-plugins/otel-trace-group-prepper/src/main/java/com/amazon/dataprepper/plugins/prepper/oteltracegroup/OTelTraceGroupPrepper.java

```diff
@@ -83,19 +85,20 @@ public OTelTraceGroupPrepper(final PluginSetting pluginSetting) {
             }
         }
 
-        final Map<String, String> traceIdToTraceGroup = searchTraceGroupByTraceIds(traceIdsToLookUp);
+        final Map<String, TraceGroupWrapper> traceIdToTraceGroup = searchTraceGroupByTraceIds(traceIdsToLookUp);
         for (final Map.Entry<Record<String>, Map<String, Object>> entry: recordMissingTraceGroupToRawSpanMap.entrySet()) {
             final Record<String> record = entry.getKey();
             final Map<String, Object> rawSpanMap = entry.getValue();
             final String traceId = (String) rawSpanMap.get(OTelTraceGroupPrepperConfig.TRACE_ID_FIELD);
-            final String traceGroup = traceIdToTraceGroup.get(traceId);
-            if (!Strings.isNullOrEmpty(traceGroup)) {
-                rawSpanMap.put(OTelTraceGroupPrepperConfig.TRACE_GROUP_FIELD, traceGroup);
+            final TraceGroupWrapper traceGroup = traceIdToTraceGroup.get(traceId);
+            if (Objects.nonNull(traceGroup)) {
                 try {
+                    Map<String, Object> traceGroupMap = OBJECT_MAPPER.convertValue(traceGroup, MAP_TYPE_REFERENCE);
+                    rawSpanMap.putAll(traceGroupMap);
                     final String newData = OBJECT_MAPPER.writeValueAsString(rawSpanMap);
                     recordsOut.add(new Record<>(newData, record.getMetadata()));
                     recordsOutFixedTraceGroupCounter.increment();
-                } catch (JsonProcessingException e) {
+                } catch (Exception e) {
                     recordsOut.add(record);
```

Comment:
Similar to the RawPrepper exception handling, do we want to instead drop the record and emit a metric instead of adding it to the output records?

Reply:
Good catch. I will emit the metric but probably not drop the record here b/c the exception is associated with the `newData` (Thus we can still output the old record).